lib/src/main/java/io/ably/lib/realtime/AblyRealtimeBase.java

```diff
@@ -97,58 +106,35 @@ protected void onAuthError(ErrorInfo errorInfo) {
         connection.connectionManager.onAuthError(errorInfo);
     }
 
-    /**
-     * A collection of Channels associated with this Ably Realtime instance.
-     */
-    public interface Channels extends ReadOnlyMap<String, Channel> {
-        /**
-         * Get the named channel; if it does not already exist,
-         * create it with default options.
-         * @param channelName the name of the channel
-         * @return the channel
-         */
-        Channel get(String channelName);
+    protected abstract RealtimeChannelBase createChannel(AblyRealtimeBase<PushType, PlatformType, ChannelType> ablyRealtime, String channelName, ChannelOptions channelOptions) throws AblyException;
 
-        /**
-         * Get the named channel and set the given options, creating it
-         * if it does not already exist.
-         * @param channelName the name of the channel
-         * @param channelOptions the options to set (null to clear options on an existing channel)
-         * @return the channel
-         * @throws AblyException
-         */
-        Channel get(String channelName, ChannelOptions channelOptions) throws AblyException;
-
-        /**
-         * Remove this channel from this AblyRealtime instance. This detaches from the channel
-         * and releases all other resources associated with the channel in this client.
-         * This silently does nothing if the channel does not already exist.
-         * @param channelName the name of the channel
-         */
-        void release(String channelName);
+    protected RestChannelBase createChannel(AblyBase ablyBase, String channelName, ChannelOptions channelOptions) throws AblyException {
+        // This method is here only due to the incremental refactoring work, AblyRealtime should never want to create
+        // an Ably REST channel. After extracting AblyRestBase from AblyBase this method should be removed.
+        return null;
```

Comment:
Could you make sure this fails if that happens by throwing an unchecked exception here? Illegal state or something like that.

Reply:
https://github.com/ably/ably-java/pull/768/commits/99825b41448ae33a16e9b1364c252308a875d273